Update overflowing int and floats to long and double

LoganElliott opened the pull request.

LoganElliott committed:
Updated int to long and float to double

LoganElliott committed:
Updated tests to match new long and double values

LoganElliott committed:
Merge pull request #1 from LoganElliott/bug/fixed-error-from-int-and-float-overflow

Bug/fixed error from int and float overflow

pitermarx commented:
what if i'm expecting that new Expression("1").Evaluate() returns an int and not a long?
Why should the default for an integer be a long and not an int?
Why should the default for a real be a double and not a float?
I'm not sure about this. Give me an example of failing code please.

In order to keep the repo faithfull to the original source i should reject this PR.
http://ncalc.codeplex.com/wikipage?title=values&referringTitle=Home

LoganElliott commented:
Note that is code is more of a suggestion of changes. Expecting an int to be the result is certainly valid.
I haven't go though and added long/double along side as that wasn't needed for my application, I replaced them as that was easiest.

int should probably be the default for an integer, personally I think double should be the default over float.

This fails 

```
var maxInt = Int32.MaxValue;
var overflow = new Expression(maxInt + "+1").Evaluate();
```

Where overflow will be equal to -2147483648

pitermarx commented:
I will close this pull request and open an issue on NCalc2 about this failing condition.
In my oppinion the NCalc engine should work as closely to the .net numeric operations.
So to have a (long) calculation there should be a way to "cast" the value to a long
